src/test/java/nextstep/subway/path/application/PathServiceTest.java

```diff
@@ -0,0 +1,126 @@
+package nextstep.subway.path.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityNotFoundException;
+import nextstep.subway.line.application.LineService;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.path.dto.PathRequest;
+import nextstep.subway.path.dto.PathResponse;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.dto.StationResponse;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class PathServiceTest {
+
+    @Mock
+    LineService lineService;
+
+    @Mock
+    StationService stationService;
+
+    Station 교대역;
+    Station 강남역;
+    Station 양재역;
+    Station 남부터미널역;
+    Station 공사중인역;
+
+    Line 이호선;
+    Line 삼호선;
+    Line 신분당선;
+    List<Line> lines;
+
+    @BeforeEach
+    void setUp() {
+        교대역 = new Station("교대역");
+        강남역 = new Station("강남역");
+        양재역 = new Station("양재역");
+        남부터미널역 = new Station("남부터미널역");
+        공사중인역 = new Station("공사중인역");
+
+        이호선 = new Line("2호선", "green", 교대역, 강남역, 10);
+        삼호선 = new Line("3호선", "orange", 교대역, 남부터미널역, 5);
+        삼호선.addSection(남부터미널역, 양재역, 3);
+        신분당선 = new Line("신분당선", "orange", 강남역, 양재역, 10);
+        lines = new ArrayList<>();
+        lines.add(이호선);
+        lines.add(삼호선);
+        lines.add(신분당선);
+    }
+
+    @DisplayName("최단 경로를 구한다")
+    @Test
+    void findShortestPath() {
+        PathRequest request = new PathRequest(4L, 1L);
+        when(lineService.findAll()).thenReturn(lines);
```

Comment:
BDD(Behavior Driven Development) 스타일의 테스트 작성방법을 사용해서 given/when/then 형식의 테스트 구조가 잡혀있는 것 같은데요 :)

`stubbing` 을 BDDMockito.given()을 사용해보면 어떨까요?

```suggestion
        given(lineService.findAll()).willReturn(lines);
```

참고: https://velog.io/@lxxjn0/Mockito%EC%99%80-BDDMockito%EB%8A%94-%EB%AD%90%EA%B0%80-%EB%8B%A4%EB%A5%BC%EA%B9%8C

Reply:
동작은 갖지만 given / when/ then 에 맞게 읽기 쉽게 도와주는 역할을 하는 군요! 좋은 정보 감사합니다 :)